packages/sdk-node/src/authenticateRequest.ts

```diff
@@ -56,3 +81,26 @@ export const decorateResponseWithObservabilityHeaders = (res: ServerResponse, re
   requestState.reason && res.setHeader(constants.Headers.AuthReason, encodeURIComponent(requestState.reason));
   requestState.status && res.setHeader(constants.Headers.AuthStatus, encodeURIComponent(requestState.status));
 };
+
+const isDevelopmentFromApiKey = (apiKey: string): boolean =>
+  apiKey.startsWith('test_') || apiKey.startsWith('sk_test_');
+
+const getRequestUrl = (req: IncomingMessage): URL => {
+  return new URL(req.url as string, `${getRequestProto(req)}://${req.headers.host}`);
+};
+
+const getRequestProto = (req: IncomingMessage): string => {
+  const proto = req.headers[constants.Headers.ForwardedProto] as string;
+  if (!proto) {
+    return req.headers.host && req.headers.host.match('localhost') ? 'http' : 'https';
```

Comment:
based on the express codebase we probably could do this: 
```typescript
proto = req.connection.encrypted ? 'https' : 'http';
```
ref: https://github.com/expressjs/express/blob/0debedf4f31bb20203da0534719b9b10d6ac9a29/lib/request.js#LL307C1-L309C14

Reply:
Thanks for the suggestion @dimkl but unfortunately this check does not apply in our case.

The `req` object's type is `IncomingMessage` and the [connection property](https://nodejs.org/api/http.html#messageconnection) is different than ClientRequest's `connection` property. This is an alias for the socket.